src/snovault/elasticsearch/searches/queries.py

```diff
@@ -1000,3 +1001,20 @@ def build_query(self):
         self.add_aggregations_and_aggregation_filters()
         self.add_matrix_aggregations()
         return self.search
+
+
+class AuditMatrixQueryFactoryWithFacets(BasicMatrixQueryFactoryWithFacets):
+    '''
+    Like BasicMatrixQueryFactoryWithFacets but adds aggregation for all audit fields.
+    '''
+
+    def _get_matrix_definition_name(self):
+        return self.kwargs.get('matrix_definition_name') or AUDIT
```

Comment:
in what cases will matrix_definition_name get set?

Reply:
For example the summary view is just a matrix view with the definition set to summary: https://github.com/ENCODE-DCC/snovault/blob/32a140e46036be806321f837ba0e3cf570e918f7/src/snowflakes/search_views.py#L204

This allows you to add multiple definitions to the item type under different names that use the same underlying code.

So all you would need to do to make, say, an audit by lab (instead of assay title) view is create a new definition on the type and pass it in to the view. You get to reuse all of the backend code.